rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionBase.java

```diff
@@ -64,12 +67,15 @@ public synchronized String sendTransaction(Web3.CallArguments args) {
                 args.data = args.data.substring(2);
             }
 
-            synchronized (transactionPool) {
+            synchronized (transactionGateway) {
```

Comment:
I don't think this is correct. This method derives the account nonce based on the line right below this one. To do that consistently it needs to synchronized over the `transactionPool` because it is accessing it as you can see in the line right below this one.

Reply:
got it

Reply:
now it synchronises with txPool instead of txGateway